packages/lightning-lsp-common/src/__tests__/context.test.ts

```diff
@@ -154,6 +154,26 @@ it('isLWCJavascript()', async () => {
     expect(await context.isLWCJavascript(document)).toBeTruthy();
 });
 
+it('isLWCTypeScript()', async () => {
+    const context = new WorkspaceContext(CORE_PROJECT_ROOT);
+
+    // lwc .ts
+    let document = readAsTextDocument(CORE_PROJECT_ROOT + '/modules/one/app-nav-bar/app-nav-bar.ts');
+    expect(await context.isLWCTypeScript(document)).toBeTruthy();
+
+    // lwc .ts outside namespace root
+    document = readAsTextDocument(CORE_ALL_ROOT + '/ui-force-components/modules/force/input-phone/input-phone.ts');
```

Comment:
Not sure I understand why this is outside of the namespace root.

Reply:
This is to test that if a ts file that is outside of the VSCode workspace is modified, we do not update tsconfig.json. Here the context workspace root is `CORE_PROJECT_ROOT` which is `ui-global-components` so if a file in `ui-force-components` is modified the return value of `isLWCTypeScript` should return false. I'll add an inline comment to explain this.